examples/with-api-destination/main.tf

```diff
@@ -200,3 +203,48 @@ data "aws_iam_policy_document" "assume_role" {
     }
   }
 }
+
+module "kms" {
+  source      = "terraform-aws-modules/kms/aws"
+  version     = "~> 2.0"
+  description = "KMS key for EventBridge"
+
+  # Aliases
+  aliases                 = ["test"]
+  aliases_use_name_prefix = true
+  key_statements = [
+    {
+      sid = "Allow use of the key"
+      principals = [
+        {
+          type        = "AWS"
+          identifiers = ["arn:aws:iam::${data.aws_caller_identity.current.account_id}:root"]
+        }
+      ]
+      actions = [
+        "kms:DescribeKey",
+        "kms:GenerateDataKey",
+        "kms:Decrypt"
+      ]
+      resources = ["*"]
+      conditions = [
+        {
+          test     = "StringLike"
+          values   = ["secretsmanager.*.amazonaws.com"]
+          variable = "kms:ViaService"
+        },
+        {
+          test     = "StringLike"
+          values   = ["arn:aws:secretsmanager:*:*:secret:events!connection/*"]
```

Comment:
Is `events!connection` the correct value? I think I didn't see `!` being used there before.

Reply:
Its for secretsmanager managed secrets: 
- https://docs.aws.amazon.com/secretsmanager/latest/userguide/service-linked-secrets.html
- https://registry.terraform.io/providers/hashicorp/aws/5.98.0/docs/resources/cloudwatch_event_connection#example-usage-cmk-encryption
- https://docs.aws.amazon.com/eventbridge/latest/userguide/encryption-connections.html